maintenance/errors/context_test.go

```diff
@@ -0,0 +1,154 @@
+package errors
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestHide(t *testing.T) {
+	iAmAnError := errors.New("i am an error")
+	iAmAnotherError := errors.New("i am another error")
+
+	type args struct {
+		ctx        context.Context
+		err        error
+		exposedErr error
+	}
+	tests := []struct {
+		name string
+		args args
+		want error
+	}{
+		{
+			name: "normal_context_no_error_nothing_exposed",
+			args: args{
+				ctx:        context.Background(),
+				err:        iAmAnError,
+				exposedErr: nil,
+			},
+			want: iAmAnError,
+		},
+		{
+			name: "normal_context_no_error_with_exposed",
+			args: args{
+				ctx:        context.Background(),
+				err:        iAmAnError,
+				exposedErr: iAmAnotherError,
+			},
+			want: fmt.Errorf("%w: %s", iAmAnotherError, iAmAnError),
+		},
+		{
+			name: "normal_context_with_error_nothing_exposed",
+			args: args{
+				ctx:        context.Background(),
+				err:        fmt.Errorf("%s: %w", iAmAnError, context.Canceled),
+				exposedErr: nil,
+			},
+			want: fmt.Errorf("%s: %s", iAmAnError, context.Canceled),
+		},
+		{
+			name: "normal_context_with_error_with_exposed",
+			args: args{
+				ctx:        context.Background(),
+				err:        fmt.Errorf("%s: %w", iAmAnError, context.Canceled),
+				exposedErr: iAmAnotherError,
+			},
+			want: fmt.Errorf("%w: %s: %s", iAmAnotherError, iAmAnError, context.Canceled),
+		},
+		{
+			name: "canceled_context_no_error_nothing_exposed",
+			args: args{
+				ctx:        canceledContext(),
```

Comment:
calling `canceledContext` like this makes the test slow, I prefer to store the result and use it in all cases.

Reply:
Huh? This doesn't look like something that takes more than a ms combined. Or is there more to it?

Reply:
I did not want to comment on this but I'm on @Teelevision side here. With this argument we would need to replace all repeating `context.Background()` calls in all test cases.

Reply:
I will change it nevertheless...